dfdaemon/config/config.go

```diff
@@ -114,6 +114,11 @@ type Properties struct {
 	LocalIP    string          `yaml:"localIP" json:"localIP"`
 	PeerPort   int             `yaml:"peerPort" json:"peerPort"`
 	StreamMode bool            `yaml:"streamMode" json:"streamMode"`
+
+	ProtocolConf []ProtocolConfig `yaml:"protocolConf" json:"protocolConf"`
+	// Mode is the default pattern.
+	Mode        string          `yaml:"mode" json:"mode"`
+	PatternConf []PatternConfig `yaml:"patternConf" json:"patternConf"`
```

Comment:
It would be better if `PatternConf` and `PatternConfig` gets changed as well. I assume you want to say something like "dfddaemon 以 P2P 模式运行"，in that case `mode` is the appropriate word to use. `pattern` is confusing.

Reply:
Here, I want to express how to package some progresses such as download, upload, data encode/decode, schedule.  In this case, pattern will be more accurate expression.

Reply:
If `PatternConfig` is meant to be used as general plugin configration, then at least a `pattern` should be categorized. For example, `cfg.DefaultPattern` clearly means `DefaultDownloadPattern`. And even so, I'm pretty sure `pattern` doesn't mean what you want it to mean. I'd rather call it `PluginConfig` to show that it configures some customizable part of the program.

Reply:
Maybe we should call it `DownloadMode` and `DownloadConf` just like `Protocol` ?

Reply:
Hm, DownloadMode and DownloadConf may be a good advice!

Reply:
@inoc603 PTAL